pkg/topology/kademlia/kademlia.go

```diff
@@ -281,23 +281,22 @@ func (k *Kad) connectBalanced(wg *sync.WaitGroup, peerConnChan chan<- *peerConnI
 	}
 }
 
-// connectNeighbours attempts to connect to the neighbours
+// connectUntilSaturation attempts to connect to the neighbours
 // which were not considered by the connectBalanced method.
-func (k *Kad) connectNeighbours(wg *sync.WaitGroup, peerConnChan chan<- *peerConnInfo) {
+func (k *Kad) connectUntilSaturation(wg *sync.WaitGroup, peerConnChan chan<- *peerConnInfo) {
 
 	sent := 0
 	var currentPo uint8 = 0
 
-	_ = k.knownPeers.EachBinRev(func(addr swarm.Address, po uint8) (bool, bool, error) {
-		depth := k.NeighborhoodDepth()
+	_ = k.knownPeers.EachBinRev(func(addr swarm.Address, bin uint8) (bool, bool, error) {
 
-		// out of depth, skip bin
-		if po < depth {
+		// if the bin is already saturated AND the bin is out of the depth, skip bin
+		if bin < k.NeighborhoodDepth() && k.connectedPeers.BinSize(bin) >= saturationPeers {
```

Comment:
again, there is no notion of saturation within depth and we always want to connect to all peers